patches/0002-Add-crypto-backend-foundation.patch

```diff
@@ -1762,204 +1679,57 @@ index a7fa3370e66c82..3470372b567eba 100644
  	"crypto/md5"
  	"crypto/sha1"
  	"crypto/sha256"
-@@ -45,7 +46,13 @@ func pHash(result, secret, seed []byte, hash func() hash.Hash) {
+@@ -44,8 +45,37 @@ func pHash(result, secret, seed []byte, hash func() hash.Hash) {
+ 	}
  }
  
++type boringPRFError struct {
++	err error
++}
++
++func (e *boringPRFError) Error() string {
++	return e.err.Error()
++}
++
++// recoverFromBoringPRFError recovers from a panic caused by the boring backend.
++// It returns the error if it was a boringPRFError, or panics if the panic was
++// caused by something else.
++func recoverFromBoringPRFError() error {
++	if p := recover(); p != nil {
++		if err, ok := p.(boringPRFError); ok {
++			// Could happen, for example, if the seed is too large. The Go implementation doesn't limit the seed size,
++			// as RFC 5705 doesn't specify a limit, but stock OpenSSL restrict it to 1024 and CNG to 256.
++			return err.err
++		}
++		panic(p)
++	}
++	return nil
++}
++
  // prf10 implements the TLS 1.0 pseudo-random function, as defined in RFC 2246, Section 5.
--func prf10(result, secret, label, seed []byte) {
-+func prf10(result, secret, label, seed []byte) error {
-+	if boring.Enabled && boring.SupportsTLS1PRF() && boring.SupportsHash(crypto.MD5SHA1) {
+ func prf10(result, secret, label, seed []byte) {
++	if boring.Enabled && boring.SupportsTLS1PRF() {
 +		if err := boring.TLS1PRF(result, secret, label, seed, nil); err != nil {
-+			return fmt.Errorf("crypto/tls: prf10: %v", err)
++			panic(boringPRFError{fmt.Errorf("crypto/tls: prf10: %v", err)})
```

Comment:
Could wrap this in a func that assembles the struct and panics (kind of symmetric to `recoverFromBoringPRFError`), but this side is pretty simple compared to the recover, so I'm indifferent. 😄

Reply:
A good suggestion is a good suggestion. Will do 👍.